android/app/src/main/java/app/covidshield/receiver/worker/ExposureCheckSchedulerWorker.kt

```diff
@@ -8,17 +8,29 @@ import app.covidshield.extensions.log
 import com.facebook.react.ReactApplication
 import com.facebook.react.bridge.ReactContext
 import com.facebook.react.modules.core.RCTNativeAppEventEmitter
+import com.google.android.gms.nearby.Nearby
+import com.google.android.gms.nearby.exposurenotification.ExposureNotificationStatus
 import kotlinx.coroutines.*
+import kotlinx.coroutines.tasks.await
 
 private const val HEADLESS_JS_TASK_NAME = "EXPOSURE_CHECK_HEADLESS_TASK"
 private const val HEADLESS_JS_TASK_TIMEOUT_MS = 60000L
 
 class ExposureCheckSchedulerWorker (val context: Context, parameters: WorkerParameters) :
         CoroutineWorker(context, parameters) {
 
+    private val exposureNotificationClient by lazy {
+        Nearby.getExposureNotificationClient(context)
+    }
+
     override suspend fun doWork(): Result {
         Log.d("background", "ExposureCheckSchedulerWorker - doWork")
         try {
+            val enStatus = exposureNotificationClient.status.await()
+            if (!enStatus.contains(ExposureNotificationStatus.ACTIVATED)){
```

Comment:
Should we use `INACTIVATED` here?

> Exposure notification is not running

That would avoid checking other possible states i.e. the UNKNOWN status etc...

Reply:
This use isEnabled here

Found this:

https://developers.google.cn/android/exposure-notifications/implementation-guide?hl=zh-cn#doWork

```

    override suspend fun doWork(): Result {
       ...

        return try {
            // Skip work if API is not enabled or the Worker has been stopped.
            if (isEnabled && !isStopped()) {
              ...
```

Reply:
Also noting for isEnabled

> Note: This method does not tell you whether the system is actually scanning or not. In other words, this returns true even if BT is off.

Reply:
@timarney - strangely enough, the reference app from Google doesn't seem to implement `isDisabled` at all.

Reply:
Looking to use `isEnabled`

Reply:
Ha ha! I meant to type `isEnabled` in my comment above.  ;)